### PR TITLE
add search by title

### PR DESCRIPTION
**Pull request by HadeerMagdy17**

### CODE REVIEW

File: src/Components/Projects/Projects.tsx [hunk 1]
```diff
@@ -164,6 +168,22 @@ const Projects: React.FC = () => {
...
   };
   // *****************************************************
+  // **********search by proj name***********************
+  const getProjectTitleValue = (e: MouseEvent) => {
+    getAllProjectsList(1, e.target.value);
+    setSearchString(e.target.value);
+  };
+  // ************************
+
+  useEffect(() => {
```
Comment (Ahmed-Abou-Emran) [inline src/Components/Projects/Projects.tsx:177]: Debouncing is in place, but there's a small glitch: when checking the network tab, you'll notice a network request being triggered with every keystroke. 
Try figuring out what is going wrong.

debouncing

Comment (Ahmed-Abou-Emran) [reply src/Components/Projects/Projects.tsx:177]: If you encounter any challenges figuring out what is going wrong, feel free to reach out to @Rawan-Mohamed , as she has experience dealing with a similar issue.

**Review by Ahmed-Abou-Emran — CHANGES_REQUESTED** [review]
### Required Changes

- [x]  Fixing debouncing not working bug

File: src/Components/Projects/Projects.tsx (hunk 1 above)
Comment (HadeerMagdy17) [reply src/Components/Projects/Projects.tsx:177]: done

**Review by Ahmed-Abou-Emran — APPROVED** [review]

**Pull request merged**